tests/Testcontainers.Tests/Fixtures/Images/DockerIgnoreFileFixture.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace DotNet.Testcontainers.Tests.Fixtures
+{
+  using DotNet.Testcontainers.Configurations;
+  using DotNet.Testcontainers.Images;
+  using Xunit;
+
+  public class DockerIgnoreFileFixture : TheoryData<IgnoreFile, string, bool>
+  {
+    public DockerIgnoreFileFixture()
+    {
+      var logger = TestcontainersSettings.Logger;
+
+      var ignoreFileDoesNotExist = new DockerIgnoreFile("/mypath/", ".dockerignore", logger, new FileSystemFixture()
+      {
+        Exists = false,
+      });
+
+      var ignoreDirectory = new DockerIgnoreFile("/mypath/", ".dockerignore", logger, new FileSystemFixture()
+      {
+        Exists = true,
+        DirectoryFullName = "/mypath",
+        Lines =
+        {
+          "bin/",
+        },
+      });
+
+      var ignoreDirectoryDockerIgnoreAndDockerFile = new DockerIgnoreFile("/mypath/", ".dockerignore", logger, new FileSystemFixture()
+      {
+        Lines =
+        {
+          "bin/",
+          ".dockerignore",
+          "Dockerfile",
+        },
+        Exists = true,
+        DirectoryFullName = "/mypath",
+      });
+
+      var ignoreDirectoryWildcardDockerIgnoreAndWildcardDockerFile = new DockerIgnoreFile("/mypath/", ".dockerignore", logger, new FileSystemFixture()
+      {
+        Lines =
+        {
+          "**/bin/",
+          "**/.dockerignore",
+          "**/Dockerfile",
+        },
+        Exists = true,
+        DirectoryFullName = "/mypath",
+      });
+
+      var dockerIgnoreFile = ".dockerignore";
+      var dockerfile = "Dockerfile";
+      var nestedDockerfile = "/lipsum/lorem/Dockerfile";
+      var binDirectory = "bin";
+
+      this.Add(ignoreFileDoesNotExist, dockerIgnoreFile, true);
+      this.Add(ignoreFileDoesNotExist, dockerfile, true);
+      this.Add(ignoreFileDoesNotExist, nestedDockerfile, true);
+      this.Add(ignoreFileDoesNotExist, binDirectory, true);
+
+      this.Add(ignoreDirectory, dockerIgnoreFile, true);
+      this.Add(ignoreDirectory, dockerfile, true);
+      this.Add(ignoreDirectory, nestedDockerfile, true);
+      this.Add(ignoreDirectory, binDirectory, false);
+
+      this.Add(ignoreDirectoryDockerIgnoreAndDockerFile, dockerIgnoreFile, true);
+      this.Add(ignoreDirectoryDockerIgnoreAndDockerFile, dockerfile, true);
+      this.Add(ignoreDirectoryDockerIgnoreAndDockerFile, nestedDockerfile, true);
+      this.Add(ignoreDirectoryDockerIgnoreAndDockerFile, binDirectory, false);
+
+      this.Add(ignoreDirectoryWildcardDockerIgnoreAndWildcardDockerFile, dockerIgnoreFile, true);
+      this.Add(ignoreDirectoryWildcardDockerIgnoreAndWildcardDockerFile, dockerfile, true);
+      this.Add(ignoreDirectoryWildcardDockerIgnoreAndWildcardDockerFile, nestedDockerfile, true);
```

Comment:
I think you expect a wrong behavior here. The `IgnoreFile` expects all paths within the context. It does not make sense to ignore (accept / deny) files that are not within the context (the implementation sees the paths as part of `/mypath`). Without double-checking the code, I expect ignoring this file is correct.